scripts/upload-ami

```diff
@@ -61,6 +99,40 @@ def upload_to_cloud(buildid, region, bucket, suffix=None, public=False):
         print("Failed to build/upload image! Error: {}".format(upload_err))
         raise
 
+    # If the Aliyun image isn't public, grant access to CI/DEV/QE accounts
+    # TODO: add `--grant-user` support to Aliyun in `ore`
+    if not public and cloud == ALIYUN:
+        arch = subprocess.check_output(["cosa", "basearch"], text='UTF-8').strip()
+        image_id = find_image_id(cloud, region, buildid, arch)
+        share_cmd = ['aliyun', '--endpoint', get_aliyun_endpoint(region), 'ecs',
+                      'ModifyImageSharePermission', '--region', region,
+                      '--ImageId', image_id]
+
+        # The aliyun CLI will return an error if you try to share an image with
+        # your own account, so this is a hack to prevent that from happening
+        # in a RHCOS devel pipeline scenario
+        aliyun_accts = [OPENSHIFT_ALIYUN_CI, OPENSHIFT_ALIYUN_DEV, OPENSHIFT_ALIYUN_QE]
+        whoami_cmd = ['aliyun', 'sts', 'GetCallerIdentity']
+        out = subprocess.check_output(whoami_cmd, text='UTF-8')
+        whoami_dict = json.loads(out)
+        whoami_id = whoami_dict['AccountId']
+
+        # And since you have to increment the number for the `--AddAcccount.n`
+        # when adding multiple accounts, we have to increment a counter while
+        # searching if our account is in the list of Aliyun accounts.
+        account_num = 0
+        for acct in aliyun_accts:
+            if acct != whoami_id:
+                account_num += 1
+                share_cmd.extend(['--AddAccount.{}'.format(account_num), acct])
+
+        try:
+            print(share_cmd)
+            subprocess.check_call(share_cmd)
+        except subprocess.CalledProcessError as upload_err:
+            print("Failed to grant access to image! Error: {}".format(upload_err))
+            raise
```

Comment:
I think *both* printing and re-throwing errors is an anti-pattern.  Instead perhaps:

```suggestion
            raise SystemExit(f"Failed to grant access to image: {upload_err}")
```

A bit like https://github.com/coreos/coreos-assembler/blob/6ac16f68b8842ca853a8b1dd41a2e38ee3095dac/src/cmd-build-validate#L50

Reply:
(here and other places)